### PR TITLE
migrate html_add_permalinks=None and html_add_permalinks=""

### PR DESCRIPTION
**Pull request by graingert**
The docs list: Set it to None or the empty string to disable permalinks.

Subject: migrate html_add_permalinks=None and html_add_permalinks=""

### Feature or Bugfix

- Bugfix

### Purpose
setting  html_add_permalinks="" no longer disables permalinks as documented

### Detail
- <feature1 or bug1>
- <feature2 or bug2>

### Relates
- <URL or Ticket>

### CODE REVIEW

File: sphinx/builders/html/__init__.py [hunk 1]
```diff
@@ -1221,15 +1221,26 @@ def validate_html_favicon(app: Sphinx, config: Config) -> None:
...
-        else:
-            config.html_permalinks_icon = html.escape(config.html_add_permalinks)  # type: ignore  # NOQA
+    html_add_permalinks = config.html_add_permalinks
+    if html_add_permalinks is UNSET:
+        return
+
+    if (
+        html_add_permalinks is None or
+        html_add_permalinks is False or
+        html_add_permalinks == ""
```
Comment (tk0miya) [inline sphinx/builders/html/__init__.py]: It seems this is equivalent to `not html_add_permalinks`. How about using it instead?

**Review by tk0miya — CHANGES_REQUESTED** [review]
Thank you for your work. Indeed, this must be handled. I'd like to merge this into 4.0 release. So could you rebase this into master branch instead?

File: sphinx/builders/html/__init__.py [hunk 2]
```diff
@@ -1248,18 +1248,29 @@ def validate_html_favicon(app: Sphinx, config: Config) -> None:
...
+        return
+
+    # RemovedInSphinx60Warning
+    logger.warning(__('html_add_permalinks has been deprecated since v3.5.0. '
+                      'Please use html_permalinks and html_permalinks_icon instead.'))
+    if (
+        html_add_permalinks is None or
+        html_add_permalinks is False or
+        html_add_permalinks == ""
+    ):
```
Comment (graingert) [inline sphinx/builders/html/__init__.py]:
```suggestion
    if not html_add_permalinks:
```

**Comment by graingert** [comment]
@tk0miya retargeted

File: sphinx/builders/html/__init__.py [hunk 3]
```diff
@@ -1248,18 +1248,25 @@ def validate_html_favicon(app: Sphinx, config: Config) -> None:
...
+
+    # RemovedInSphinx60Warning
+    logger.warning(__('html_add_permalinks has been deprecated since v3.5.0. '
+                      'Please use html_permalinks and html_permalinks_icon instead.'))
+    if not html_add_permalinks:
+        config.html_permalinks = False  # type: ignore[attr-defined]
+        return
 
+    config.html_permalinks_icon = html.escape(  # type: ignore[attr-defined]
+        config.html_add_permalinks
```
Comment (graingert) [inline sphinx/builders/html/__init__.py]:
```suggestion
        html_add_permalinks
```

**Review by tk0miya — APPROVED** [review]

**Comment by tk0miya** [comment]
Thank you for quick update. Merging!

**Comment by graingert** [comment]
@tk0miya will you cherry pick this to the v3 branch?

**Comment by tk0miya** [comment]
No, I don't have a plan to release 3.6 now. So it's no longer needed.

**Pull request merged**